azkaban-common/src/main/java/azkaban/executor/ExecutorApiClient.java

```diff
@@ -182,10 +183,19 @@ public boolean isReverseProxyEnabled() {
    *
    * @return http client
    */
-  protected CloseableHttpClient createHttpsClient() {
+  protected CloseableHttpClient createHttpsClient(final Optional<Integer> httpTimeout) {
     final HttpClientBuilder httpClientBuilder = HttpClients.custom()
         .setSSLSocketFactory(this.tlsSocketFactory);
-    return httpClientBuilder.build();
+    if (httpTimeout.isPresent()) {
+      int timeout = httpTimeout.get();
+      final RequestConfig requestConfig = RequestConfig.custom()
+          .setConnectTimeout(timeout)
+          .setSocketTimeout(timeout)
+          .build();
+      return httpClientBuilder.setDefaultRequestConfig(requestConfig).build();
+    } else {
+      return httpClientBuilder.create().build();
```

Comment:
`create()` method was called by the `custom()` invocation a few lines above. Please don't call `create()` again here.

Reply:
ack